### DISCUSSION TITLE
While waiting for features to protect some traits, is the following a good way to protect them?

### DISCUSSION THREAD

**Discussion by frederikhors** [ory/kratos#2366] [Q&A] [answered]
_Thank you for the tremendous work you have done and continue to do every day for this project. You are wonderful and you will be rewarded with gold because commitment and dedication always pay off. Always._

I can't figure out (but it's my fault) if disabling in the `kratos.yml` config file the `selfservice_methods_profile` with the below lines I can rest assured that no identity will ever be able to change it's own traits.

```yml
selfservice:
  methods:
    profile:
      enabled: false
```

Since we have company traits that cannot be changed by the identity and since we don't care if the identity can read those traits, _while waiting for the features (like https://github.com/ory/kratos/issues/47) to arrive_, we have decided to "protect" the change in this way.

1. Can we sleep peacefully like this?

1. Can we still use admin commands to change those traits?

**Comment by aeneasr** [comment, marked as answer]
Hey @frederikhors , glad to see you are still around :) Yes, that is for sure a safe bet! An alternative would be to store the data you want to modify differently in another database outside of Kratos!

**Reply by frederikhors** [reply]
I know we could use another DB, but these traits are the ones necessary for EVERY call to check role and other things like that.

So each call would be "burdened" by the first round trip for DB Kratos and the second round trip for DB traits.

I think this is a common problem.

What do you think?

**Reply by aeneasr** [reply]
That makes sense! We want to introduce a new field called `metadata` that can be read by the user, but not modified by them. Would that be helpful?

**Reply by frederikhors** [reply]
Yep. Something like that. Not necessarily public too.

**Reply by aeneasr** [reply]
We had an idea to have "admin" and "public" metadata. "admin" metadata would only be available when doing `GET /admin/identities/...` and public metadata would also be available at `/session/whoami`. WDYT?

**Reply by frederikhors** [reply]
Ok. I think it's fine.

**Reply by woylie** [reply]
For reference: #117
